### ISSUE TITLE
Chain crashing at height 2016

### ISSUE THREAD

**Issue by lgalabru** [blockstack/stacks-blockchain#1503]
**Describe the bug**
Neon is rebooting every 3 to 4 hours. Looking at the logs, it looks like there's an issue at block height 2016, we are constantly crashing with the following exception:

```

Apr 25 21:33:13 public-regtest-002 docker[330820]: INFO [1587850393.405] [testnet/src/neon_node.rs:307] Won sortition! stacks_header=7a020d09460a56a28754d9fe4ea11693a2fa7b985d61b5b1b3dde0ed394103c3, burn_header=32ee31f2ec9c4567e1d172199f2c
Apr 25 21:33:15 public-regtest-002 docker[330820]: INFO [1587850395.503] [src/chainstate/stacks/miner.rs:276] Miner: mined anchored block 0b9726ea51984ba5c1175d13169c39ad1aff6754af7a53e41a5e0b09237a9b05, parent block 7a020d09460a56a28754d9
Apr 25 21:33:15 public-regtest-002 docker[330820]: INFO [1587850395.520] [testnet/src/neon_node.rs:675] Finish tenure: 0b9726ea51984ba5c1175d13169c39ad1aff6754af7a53e41a5e0b09237a9b05
Apr 25 21:33:20 public-regtest-002 docker[330820]: ERROR [1587850400.156] [src/burnchains/bitcoin/spv.rs:152] bits mismatch at block 2016 of /tmp/stacks-testnet-550584ab5573eb17/burnchain/spv-headers.dat.reorg (offset 0 interval 1 of 1-2):
Apr 25 21:33:20 public-regtest-002 docker[330820]: ERROR [1587850400.158] [src/burnchains/bitcoin/spv.rs:257] Received headers with bad target or difficulty
Apr 25 21:33:20 public-regtest-002 docker[330820]: WARN [1587850400.159] [src/burnchains/bitcoin/indexer.rs:414] Unhandled error InvalidPoW
Apr 25 21:33:20 public-regtest-002 docker[330820]: ERROR [1587850400.159] [src/burnchains/bitcoin/indexer.rs:508] Failed to fetch headers from 2000 - 2016
Apr 25 21:33:20 public-regtest-002 docker[330820]: ERROR [1587850400.159] [src/burnchains/burnchain.rs:794] Failed to check for reorgs from 2016
Apr 25 21:33:20 public-regtest-002 docker[330820]: ERROR [1587850400.159] [testnet/src/burnchains/bitcoin_regtest_controller.rs:138] Unable to sync burnchain: Invalid proof of work
Apr 25 21:33:20 public-regtest-002 docker[330820]: thread 'main' panicked at 'explicit panic', testnet/src/burnchains/bitcoin_regtest_controller.rs:139:17

```

**Steps To Reproduce**
Mine chain up to block #2016

**Expected behavior**
Chain should make progress

**Environment (please complete the following information):**
 Any

**Comment by diwakergupta** [comment]
Duplicate of #1488

**Comment by jcnelson** [comment]
Fixed by #1522
